File: python_kata/src/algorithms/sorting/bubble_sort.py
def bubble_sort(arr: list[int]) -> list[int]:
    for i in range(len(arr)):
        for j in range(0, len(arr) - 1 - i, 1):
            if arr[j] > arr[j + 1]:
                temp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = temp

    return arr

File: typescript-kata/src/algorithms/sorting/bubble-sort.ts
export function BubbleSort(arr: number[]): number[] {
    for (let i = 0; i < arr.length; ++i) {
        // we subtract here 1 because we add 1 while swapping
        // we subtract here i because on each iteration the last number is the biggest (and on first i === 0)
        for (let j = 0; j < arr.length - 1 - i; ++j) {
            if (arr[j] >= arr[j + 1]) {
                // swap(arr[j], arr[j + 1])
                const temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
    return arr;
}

function swap(val1: number, val2: number): void {
    const temp = val1;

    val1 = val2;
    val2 = temp;
}
